addons/sale_project/models/project_project.py

```diff
@@ -49,6 +49,8 @@ def _domain_sale_line_id(self):
     @api.model
     def default_get(self, fields):
         defaults = super().default_get(fields)
+        if self.env.context.get("from_sale_order_action"):
+            defaults['allow_billable'] = True
```

Comment:
why not adding `default_allow_billable: True` in the context of the action?

Reply:
> why not adding `default_allow_billable: True` in the context of the action?

We already passed it, but it didn’t work. During the opening of the project form view, the default_.. context was cleared. That’s why I did this.
here - https://github.com/odoo/odoo/blob/master/addons/sale_project/models/sale_order.py#L187

Reply:
Where is it cleared?

Reply:
> why not adding `default_allow_billable: True` in the context of the action?

Also, I have already mentioned this in the commit.

Reply:
Ah it is replaced by the project template, okay I understood

Reply:
> > > why not adding `default_allow_billable: True` in the context of the action?
> > 
> > 
> > Also, I have already mentioned this in the commit.
> 
> I read it but it does not explain why it is cleared. I checked the code and I don't see why it is cleared


In the Create a Project action, we pass default_allow_billable as false [link](https://github.com/odoo/odoo/blob/19.0/addons/project/views/project_project_views.xml#L347)
. Therefore, the default_allow_billable context passed to the sale_project will be overridden, so I have handled it based on the from_sale_order_action context